### PR TITLE
invoices/invoiceregistry: generate random payaddr for keysend

### PR DESCRIPTION
**Pull request by cfromknecht**
This commit automatically generates a random payment address for any new
keysend payments. With the addition of the payment address index, we now
enforce uniqueness of payment addresses across all invoices. As a
result, if you run master currently you can only accept one keysend
payment since they all share a zero-value payment address.

Follow up from #4285

### CODE REVIEW

File: invoices/invoiceregistry_test.go [hunk 1]
```diff
@@ -947,3 +947,50 @@ func TestInvoiceExpiryWithRegistry(t *testing.T) {
...
+	}
+
+	// Inserting a new keysend payment should succeed.
+	err := ctx.registry.processKeySend(keysendCtx)
+	assert.Nil(t, err)
+
+	// Replaying the same context should also succeed to ensure idempotency.
+	// This should be handled later in the pipeline by comparing the circuit
+	// keys.
+	err = ctx.registry.processKeySend(keysendCtx)
```
Comment (joostjager) [inline invoices/invoiceregistry_test.go:982]: I think that testing on the external interface of invoice registry is more robust.

Comment (cfromknecht) [reply invoices/invoiceregistry_test.go:982]: my thought here was that this is a better place to assert the behavior of `processKeySend` wrt to how it filters errors

**Comment by cfromknecht** [comment]
@joostjager rebased on #4334

File: invoices/invoiceregistry_test.go (hunk 1 above)
Comment (joostjager) [reply invoices/invoiceregistry_test.go:982]: Why is it better? We also already have a parameterized `TestKeysend` that tests on the external interface. Maybe that can be extended easily.

Comment (cfromknecht) [reply invoices/invoiceregistry_test.go:982]: afaict you can't test the idempotency of `processKeySend` because it settles the keysend immediately after

Comment (cfromknecht) [reply invoices/invoiceregistry_test.go]: moved everything to a white box test, but we are discussing holding off on this change and fixing as part of #4338

File: invoices/invoiceregistry.go [hunk 2]
```diff
@@ -676,13 +677,24 @@ func (i *InvoiceRegistry) processKeySend(ctx invoiceUpdateCtx) error {
 		return errors.New("final expiry too soon")
 	}
 
+	// Generate a random payment address for this invoice. The invoice
+	// databse is indexed by payment address, so we must ensure that the
```
Comment (alexbosworth) [inline invoices/invoiceregistry.go:681]:
```suggestion
	// database is indexed by payment address, so we must ensure that the
```

**Comment by cfromknecht** [comment]
Rolled into #4338

File: invoices/invoiceregistry.go (hunk 2 above)
Comment (cfromknecht) [reply invoices/invoiceregistry.go:681]: fixed in #4338

**Pull request closed without merging**